pkg/git/git_test.go

```diff
@@ -0,0 +1,100 @@
+package git_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/pctl/pkg/git"
+	"github.com/weaveworks/pctl/pkg/git/fakes"
+)
+
+var _ = Describe("git", func() {
+	var (
+		runner *fakes.FakeRunner
+	)
+
+	BeforeEach(func() {
+		runner = new(fakes.FakeRunner)
+	})
+
+	When("normal flow operations", func() {
+		It("can add changes to a commit", func() {
+			g := git.NewCLIGit(git.CLIGitConfig{
+				Location: "location",
+				Branch:   "main",
+				Remote:   "origin",
+			}, runner)
+			err := g.Add()
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("commit changes", func() {
+			g := git.NewCLIGit(git.CLIGitConfig{
+				Location: "location",
+				Branch:   "main",
+				Remote:   "origin",
+			}, runner)
+			err := g.Commit()
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("push changes to a remote", func() {
+			g := git.NewCLIGit(git.CLIGitConfig{
+				Location: "location",
+				Branch:   "main",
+				Remote:   "origin",
+			}, runner)
+			err := g.Push()
+			Expect(err).NotTo(HaveOccurred())
+		})
```

Comment:
these tests are sort of half baked, just because it didn't error doesn't mean it did what I want. Can we check somehow that they actually do what we intend?

Reply:
Jake, ignore the tests. :D I'm writing them. :D I'm just getting familiar with counterfeit. :)

Reply:
ahhh sorry!